onnxruntime/core/framework/execution_providers.h

```diff
@@ -22,7 +22,7 @@ class ExecutionProviders {
  public:
   ExecutionProviders() = default;
 
-  common::Status Add(const std::string& provider_id, std::unique_ptr<IExecutionProvider> p_exec_provider) {
+  common::Status Add(const std::string& provider_id, std::shared_ptr<IExecutionProvider> p_exec_provider) {
```

Comment:
>std::shared_ptr< [](http://example.com/codeflow?start=53&length=16)

I think the usage of shared_ptr is discouraged. And this type of usage is even worse, as it involves refcounting. You could have done some moves here or take it by ref.
In the meeting with Teams recently shared_ptr came up again.